Used go-gh table printer for listing caches

t-dedah opened the pull request:
### What are you trying to accomplish?

Removing our printing logic with go-gh table printer

### What approach did you choose and why?
### Anything you want to highlight for special attention from reviewers?
No

kotewar reviewed (approved):
LGTM

kotewar commented:
Do we release v1.0.1 with this table printer?

t-dedah commented:
yes we should release a new version but we might pick up other minor changes so finalising those with Aparna.

bishal-pdMSFT commented on internal/utils.go:
```diff
@@ -81,23 +76,6 @@ func lastAccessedTime(lastAccessedAt string) string {
 	return fmt.Sprintf(" %s", lastAccessed.FromNow())
```
Dio we need this space `" %s"` now?

t-dedah replied:
Not needed anymore, removed

bishal-pdMSFT commented on internal/utils.go:
```diff
@@ -51,17 +44,19 @@ func FormatCacheSize(size_in_bytes float64) string {
+	_ = tp.Render()
 }
 
 func PrettyPrintTrimmedCacheList(caches []types.ActionsCache) {
```
Is this method still used?

bishal-pdMSFT replied:
If not then probably CI should have thrown an error about unused code. Can we enable that if go build allows this feature?

bishal-pdMSFT replied:
Ah yes

bishal-pdMSFT commented:
@t-dedah can you please paste some screenshots at different shell window sizes to help visualize. We can also add @N-Usha and @heyamie to check those.

t-dedah commented:
Hi @bishal-pdMSFT @heyamie @N-Usha we have removed our printing logic and updated the code to reuse go-gh printer which is based on gh cli. Screenshots with latest changes:-

## Full width for the terminal

<img width="1132" alt="Screenshot 2022-08-09 at 4 21 27 PM" src="https://user-images.githubusercontent.com/59823596/183600957-ddd37dee-0808-46cf-879d-26680bf020ff.png">

## Resizing the terminal with last printer result

<img width="825" alt="Screenshot 2022-08-09 at 4 21 51 PM" src="https://user-images.githubusercontent.com/59823596/183600981-15a42498-6d20-4ddb-840d-23f3d4d72531.png">

## Printed after reducing the size of the window

<img width="810" alt="Screenshot 2022-08-09 at 4 22 05 PM" src="https://user-images.githubusercontent.com/59823596/183600984-240e6851-a744-4f0b-8657-46b92e7f7ddb.png">

## Further reduction
<img width="609" alt="Screenshot 2022-08-09 at 4 22 23 PM" src="https://user-images.githubusercontent.com/59823596/183600988-ce0b2cc2-886a-4c34-a7e7-dca38e3a2a57.png">

bishal-pdMSFT commented:
Overall looks great. I am a bit worried about the key getting truncated a bit more larger terminal size than probably it was earlier. Basically we are still not handling #21

bishal-pdMSFT commented:
> Overall looks great. I am a bit worried about the key getting truncated a bit more larger terminal size than probably it was earlier. Basically we are still not handling #21

As per discussion with @t-dedah the idea here is to enable full key visualization by using `--json` option. That will be a separate PR.
